robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,204 @@
+import logging
+
+import requests
+from tenacity import retry
+from tenacity import stop_after_attempt
+from tenacity import wait_fixed
+
+from robottelo.config import settings
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ReportPortal:
+    """Represents Report Portal in ReportPortal
+
+    This holds the properties and function to interact with the Report Portal
+    """
+
+    def __init__(self):
+        """Configure the settings and initiate report portal properties"""
+        if not settings.configured:
+            settings.configure()
+        self.rp_url = settings.report_portal.rp_url
+        self.rp_project = settings.report_portal.rp_project
+        self.rp_api_key = settings.report_portal.rp_key
+
+    @property
+    def api_url(self):
+        """Super url of report portal to be used as a base in all child classes
+        :returns Base url for API request
+        """
+        _api_url = f'{self.rp_url}/api/v1/{self.rp_project}/'
+        return _api_url
+
+    @property
+    def headers(self):
+        """The headers for Report Requests.
+        :returns header for API request
+        """
+        _headers = {'Authorization': f'Bearer {self.rp_api_key}'}
+        return _headers
+
+
+class Launch(ReportPortal):
+    """Represents Launch in Report Portal
+
+    This holds the properties and function to interact with the
+    Report Portal launches or specific launch
+    """
+
+    def __init__(self):
+        """Initiate report portal launch properties"""
+        super(Launch, self).__init__()
+        self.launch_url = f'{self.api_url}/launch'
+
+    @property
+    def params(self):
+        """Parameters for Launch API request
+
+        :returns dict: parameters for API launch request
+        """
+        _params = {
+            'page.page': 1,
+            'page.size': 500,
+            'page.sort': 'start_time',
+        }
+        return _params
+
+    @property
+    def launches(self):
+        """Returns All launches in Report Portal
+
+        This does not includes each tests data for all tests in all launches.
+        But it includes just an overview data for all and each launch in Report Portal
+
+        :returns list: List of all launches in Report Portal
+        """
+        resp = requests.get(
+            self.launch_url, headers=self.headers, params=self.params, verify=False
+        )
+        resp.raise_for_status()
+        data = resp.json()
+        if data['content']:
+            launches = sorted(data['content'], key=lambda x: x['start_time'], reverse=True)
+            return launches
+
+    def launch(self, latest):
```

Comment:
`s/launch/get_launch`
- anyway, is this method required?
You could easily use the `get_launches` method if you made it accept custom parameters.
Passing custom `page.sort` and `page.size` could easily give you access to what you want. Isn't this too overengineered?

Reply:
Probably it might look over-engineered, but it's just for simplicity.

The clear thought behind that is to differentiate get `all launches` and single `launch`. Later both will be populated with their submodules, e.g :

`Launches`: mode, names, owners, tags
`Launch`: report (probably less likely to implement because it just exports the results for a launch, but we might have more detail in the report in just getting the launch from launches)

Reply:
Also. I dont like using these verbs along with nouns. The docstrings are good enough to understand what the def is doing.